models/PyCryptoBot.py

```diff
@@ -1023,50 +1034,86 @@ def startApp(self, account, last_action='', banner=True):
+                        if self.smart_switch == 1:
+                            tradingData = self.getSmartSwitchHistoricalDataChained(self.market, 3600,
+                                                                                   str(startDate),
+                                                                                   str(endDate),
+                                                                                   str(startDate))
+
+                        else:
+                            tradingData = self.getHistoricalData(self.getMarket(), self.getGranularity(),
+                                                                startDate.isoformat(timespec='milliseconds'),
+                                                                endDate.isoformat(timespec='milliseconds'))
                         attempts += 1
                 elif self.simstartdate is not None and self.simenddate is None:
-                    date = self.simstartdate.split('-')
-                    startDate = datetime(int(date[0]), int(date[1]), int(date[2]))
+                    #date = self.simstartdate.split('-')
+                    startDate = self.getDateFromISO8601Str(self.simstartdate.replace("T", " "))
                     endDate = startDate + timedelta(minutes=(self.getGranularity()/60)*300)
-                    while len(tradingData) != 300 and attempts < 10:
-                        tradingData = self.getHistoricalData(self.getMarket(), self.getGranularity(),
+					
+                    while len(tradingData) < 300 and attempts < 10:
+                        if self.smart_switch == 1:
+                            tradingData = self.getSmartSwitchHistoricalDataChained(self.market, self.getGranularity(),
+                                                                                   str(startDate),
+                                                                                   str(endDate),
+                                                                                   str(startDate))
+
+                        else:
+                            tradingData = self.getHistoricalData(self.getMarket(), self.getGranularity(),
                                                              startDate.isoformat(timespec='milliseconds'),
                                                              endDate.isoformat(timespec='milliseconds'))
                         attempts += 1
                 elif self.simenddate is not None and self.simstartdate is None:
                     if self.simenddate == 'now':
-                        endDate = datetime.now()
+                        endDate = self.getDateFromISO8601Str(str(datetime.now()).split(".")[0])
                     else:
-                        date = self.simenddate.split('-')
-                        endDate = datetime(int(date[0]), int(date[1]), int(date[2]))
+                        endDate = self.getDateFromISO8601Str(self.simenddate.replace("T", " "))
```

Comment:
all `replace('T', ' ')` should be taken care of in the conversion function to avoid peppering the code with these `replace()` calls (and eventually missing to add this call in a few places, resulting in unstable behaviour)